src/test/java/core/basesyntax/services/FruitShopServiceImplTest.java

```diff
@@ -0,0 +1,92 @@
+package core.basesyntax.services;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.service.impl.FruitShopServiceImpl;
+import core.basesyntax.strategy.handler.OperationHandler;
+import core.basesyntax.strategy.handler.impl.BalanceHandler;
+import core.basesyntax.strategy.handler.impl.OperationStrategyImpl;
+import core.basesyntax.strategy.handler.impl.PurchaseHandler;
+import core.basesyntax.strategy.handler.impl.ReturnHandler;
+import core.basesyntax.strategy.handler.impl.SupplyHandler;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class FruitShopServiceImplTest {
+    private static final String PINEAPPLE = "pineapple";
+    private static final String BANANA = "banana";
+    private static final String ORANGE = "orange";
+    private static final String DRAGON_FRUIT = "dragonFruit";
+    private FruitShopService fruitShopService;
+
+    @BeforeEach
+    public void setUp() {
+        initFruitShopService();
+        initStorage();
+
+    }
+
+    private void initFruitShopService() {
+        Map<Operation, OperationHandler> operationHandlerMap
+                = new HashMap<>();
+        operationHandlerMap.put(Operation.BALANCE, new BalanceHandler());
+        operationHandlerMap.put(Operation.PURCHASE, new PurchaseHandler());
+        operationHandlerMap.put(Operation.RETURN, new ReturnHandler());
+        operationHandlerMap.put(Operation.SUPPLY, new SupplyHandler());
+        fruitShopService = new FruitShopServiceImpl(new OperationStrategyImpl(operationHandlerMap));
+    }
+
+    private void initStorage() {
+        Storage.storage.clear();
+        Storage.storage.put(PINEAPPLE, 10);
+        Storage.storage.put(BANANA, 20);
+        Storage.storage.put(ORANGE, 30);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        Storage.storage.clear();
```

Comment:
I think double storage.clear is redundant (initStorage and tearDown), but I`m not sure.